### Task 1.1: Dataset Summary

The dataset consist of comprehensive list of train routes and station details.

Total Records : 186074

Total Columns : 12

Column Names : SN,	Train_No,	Station_Code,	1A,	2A,	3A,	SL,	Station_Name,	Route_Number,	Arrival_time,	Departure_Time,	Distance.

### Task 1.2: Train-wise Route Table

In [1]:
import pandas as pd

# Creating the dataset
data = {
    'Train No': [107, 108, 128, 290, 401],
    'Start Station': ['SAWANTWADI R', 'MADGOAN JN.', 'MADGOAN JN.', 'DELHI-SAFDAR', 'AURANGABAD'],
    'End Station': ['MADGOAN JN.', 'SAWANTWADI R', 'CHHATRAPATI', 'DELHI-SAFDAR', 'VARANASI JN.'],
    'Total Stops': [4, 4, 22, 14, 12],
    'Max Distance (km)': [78, 83, 978, 2694, 1618]
}

# Creating the DataFrame
df = pd.DataFrame(data)

# Displaying the table
df

Train No Start Station   End Station  Total Stops  Max Distance (km)
0       107  SAWANTWADI R   MADGOAN JN.            4                 78
1       108   MADGOAN JN.  SAWANTWADI R            4                 83
2       128   MADGOAN JN.   CHHATRAPATI           22                978
3       290  DELHI-SAFDAR  DELHI-SAFDAR           14               2694
4       401    AURANGABAD  VARANASI JN.           12               1618

### Task 1.3: Basic Statistics

In [2]:
df=pd.read_csv("Dataset1.csv")
df.head()

SN  Train_No Station_Code   1A   2A   3A   SL  Station_Name  Route_Number  \
0   1       107          SWV  100  100  100  100  SAWANTWADI R             1   
1   2       107         THVM  260  228  196  164        THIVIM             1   
2   3       107         KRMI  345  296  247  198       KARMALI             1   
3   4       107          MAO  490  412  334  256   MADGOAN JN.             1   
4   1       108          MAO  100  100  100  100   MADGOAN JN.             1   

  Arrival_time Departure_Time  Distance  
0     00:00:00       10:25:00         0  
1     11:06:00       11:08:00        32  
2     11:28:00       11:30:00        49  
3     12:10:00       00:00:00        78  
4     00:00:00       20:30:00         0

In [3]:
df.describe()

SN       Train_No             1A             2A  \
count  186074.000000  186074.000000  186074.000000  186074.000000   
mean       13.914695   42157.747455    1506.769189    1225.415351   
std        12.779368   25090.080221    2418.719821    1934.975856   
min         1.000000     107.000000     100.000000     100.000000   
25%         5.000000   17225.000000     215.000000     192.000000   
50%        11.000000   40050.000000     465.000000     392.000000   
75%        18.000000   57550.000000    1555.000000    1264.000000   
max       118.000000   99908.000000   21400.000000   17140.000000   

                  3A             SL  Route_Number       Distance  
count  186074.000000  186074.000000      186074.0  186074.000000  
mean      944.061513     662.735777           1.0     281.353838  
std      1451.231892     967.524012           0.0     483.743964  
min       100.000000     100.000000           1.0       0.000000  
25%       169.000000     146.000000           1.0      23.000000  
50%       319.000000     246.000000           1.0      73.000000  
75%       973.000000     682.000000           1.0     291.000000  
max     12880.000000    8620.000000           1.0    4260.000000

#### Distance:
Mean: 281.353838 km

Median: 73 km

Maximum: 4260 km

Std. Deviation : 483.743964 km

### Task 1.4: Identify missing, null, or inconsistent values

In [12]:
df.isnull().sum()

SN                0
Train_No          0
Station_Code      0
1A                0
2A                0
3A                0
SL                0
Station_Name      0
Route_Number      0
Arrival_time      0
Departure_Time    0
Distance          0
dtype: int64

✅ Here the dataset is cleaned and structured